### PR TITLE
Create Dockerfile

### PR DESCRIPTION
**Pull request by Tylersuard**
Create dockerfile due to industry pressure.

### CODE REVIEW

**Review by CTimmerman — APPROVED** [review]
Looks ok to me.

**Comment by skyracer2012** [comment]
The cat image seems like a little bit of bloat

**Comment by Tylersuard** [comment]
I assure you it is essential.

**Comment by Vincent-bin** [comment]
looks good to me.

**Comment by hughesadam87** [comment]
Please include 30+ page analysis of container runtimes so that we can acheive preliminary stage 0.1 approval form the architecture review board.

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

**Review by Heath123 — APPROVED** [review]

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM java:8  
+
+COPY . /var/www/java  
+
+WORKDIR ./src
+
+ADD https://images.pexels.com/photos/4587959/pexels-photo-4587959.jpeg
```
Comment (PurHur) [inline dockerfile:7]: Please remove this line.

Comment (Tylersuard) [reply dockerfile:7]: I can't.  If I remove the cat picture everything stops working.

Comment (PurHur) [reply dockerfile:7]: How are we gonna resolve this issue? 

Company management wants us to scale into cloud but the cat image is blocking it.

Comment (knakamura13) [inline dockerfile:7]: Please replace the hyperlink with a locally stored `jpeg` file to appease corporate demand for cloud scalability while avoiding cross-hosting headaches as well as, God-forbid, 404 missing file issues if pexels.com goes offline.

Here is the cat image, enhanced and upscaled using Machine Learning (i.e., future-proofed for devices using 16k resolution):
https://we.tl/t-okV1rNmg3d

You should add this image to the base/root of the project and then change line 7 of the Dockerfile as follows:
```
- ADD https://images.pexels.com/photos/4587959/pexels-photo-4587959.jpeg
+ ADD cat.jpeg
```

Note that `ADD` is no longer necessary as it is the same as `COPY` when using locally-referenced files. Therefore, the same line can be rewritten as follows:
```
COPY cat.jpeg .
```

In summary, it is my expert opinion that the proposed Dockerfile in full should appear like so:
```BASH
FROM java:8  

COPY . /var/www/java  

WORKDIR ./src

COPY cat.jpeg .

RUN ./main/java/com/seriouscompany/business/java/fizzbuzz/packagenamingpackage/impl/Main.java
```

Comment (knakamura13) [reply dockerfile:7]: @PurHur please see my comment below regarding a possible resolution for this issue. https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition/pull/542/commits/bd44545cd62f4d56484b24afac18799c949fd5db#r901147477

**Review by knakamura13 — CHANGES_REQUESTED** [review]

File: dockerfile (hunk 1 above)
Comment (PurHur) [reply dockerfile:7]: Thank you very much for your work so we doing huge steps forward.

Another big problem here is the copyright of the cat image. We dont have a written contract to use this image in our software and we should avoid getting sued for 120m damages.

So i propose to replace the cat image with an ai generated cat image that has no copyright.

Comment (knakamura13) [reply dockerfile:7]: I agree that a copyright free image would be the best option here.

However, we should be cognizant to the developing field of AI-rights wherein the ownership of AI-generated works of art may belong to the AI, or perhaps the creator of said AI, and not necessarily the public domain. 

We should acquire an AI-generated cat image and revisit this issue of legality in 20 years from today's date. Put this on your calendars.

Comment (knakamura13) [reply dockerfile:7]: How about we use a cat image that I took? Copyright: me.

Artboard 2

Comment (PurHur) [reply dockerfile:7]: Can you prove the ownership and sign a contract that is added to the docker image?

Comment (knakamura13) [reply dockerfile:7]: The screenshot below shows that I have similar images taken at the same time of the same cat (I also have dozens of other pictures of this cat [his name is Kimchi 🐱]).

Screen Shot 2022-06-19 at 3 52 41 PM

I don't know if EXIF data counts as proof, but here's a screenshot of the cat image with its exif data:
Screen Shot 2022-06-19 at 3 55 15 PM

Comment (knakamura13) [reply dockerfile:7]: I'm also happy to fly out to our Enterprise Headquarters in the corporate private jet to provide additional proof.

Comment (PurHur) [reply dockerfile:7]: I guess that is enough. We should use a picture of Kimchi.

Comment (knakamura13) [reply dockerfile:7]: Now we just need @Tylersuard to update his Dockerfile and resubmit his PR.

**Comment by PurHur** [comment]
@Tylersuard do you have a time horizon for finalizing this pull request? 

Your work looks great and the copyright issue with the cat image is solved. If you could exchange the image with a picture of Kimchi that would be great. This is the last thing that blocks the Company to migrate to the cloud.

**Comment by erikhuizinga** [comment]
This opens the door to introduce a dockerfile builder, which is important, because we might want to add any number of cat pictures in the future (at least one).

**Comment by GNUGradyn** [comment]
Please include more cat breeds for localization

**Comment by PurHur** [comment]
I actually got feedback of our consulting firm. The review is positive and we got an OK for merging this! 🎉 

Who is in charge merging this great work? I guess @Mikkeren isnt working at the company anymore.

**Comment by erikhuizinga** [comment]
We clearly have issues locating a dependency (a person authorised to merge this PR). Instead of a service locator, we should try dependency injection for future PRs.

File: dockerfile (hunk 1 above)
Comment (rr-) [reply dockerfile:7]: I believe relying on pexels is not sustainable; I'd commit the image into directly the repository instead.
But I do agree we should investigate why the Dockerfile is not working unless we add more cats to it.

**Review by erhosen — APPROVED** [review]

File: dockerfile (hunk 1 above)
Comment (PurHur) [reply dockerfile:7]: >But I do agree we should investigate why the Dockerfile is not working unless we add more cats to it.

We did the research a year ago. This project needs to cut some costs so we cant do it again. One image of kimchi would not add much bloat to the current state of our goal.

**Review by mutezebra — APPROVED** [review]
